src/Microsoft.VisualStudio.Extensibility.Testing.Xunit.Shared/Threading/IdeTestCaseRunner.cs

```diff
@@ -43,16 +43,17 @@ public VisualStudioInstanceKey VisualStudioInstanceKey
 
         protected override XunitTestRunner CreateTestRunner(ITest test, IMessageBus messageBus, Type testClass, object?[] constructorArguments, MethodInfo testMethod, object?[]? testMethodArguments, string skipReason, IReadOnlyList<BeforeAfterTestAttribute> beforeAfterAttributes, ExceptionAggregator aggregator, CancellationTokenSource cancellationTokenSource)
         {
-            if (Process.GetCurrentProcess().ProcessName == "devenv")
+            if (SharedData.Exception is not null)
```

Comment:
This is related to the exception handling, but slightly speculative.  There is code in `IdeTestAssemblyRunner` to set `SharedData.Exception` when there is an exception in the harness.   It then eventually calls into this to run the test cases again to report the harness exception for each test case.

However - it appears as though if devenv is still running (aka it wasn't a devenv crash that caused this exception), then we'd never report the harness failure.

I moved this check up to report the harness failure even if we're still running devenv.

This check isn't enough by itself though - as the exception we were hitting would just be thrown again later on as we tried to report the test case starting message for the `ErrorReportingIdeTestRunner`